module/VuFindTheme/src/VuFindTheme/ResourceContainer.php

```diff
@@ -164,7 +282,28 @@ public function getCss()
      */
     public function getJs()
     {
-        return array_unique($this->js);
+        return $this->js;
+    }
+
+    /**
+     * Given a colon-delimited configuration string, break it apart, making sure
+     * that URLs in the first position are not inappropriately split.
+     *
+     * @param string $current Setting to parse
+     *
+     * @return array
+     */
+    public function parseSetting($current)
+    {
+        $parts = explode(':', $current);
```

Comment:
Could we just say that you need to use the array format in theme config if you have a protocol and condition? That would be a minor inconvenience during setup but would avoid doing this check every time.

Reply:
That sounds like the right thing to do, then we could get rid of all the string logic... i just thought that @demiankatz normally likes it if things stay backwards compatible; however we could simplify the use cases and leave a note in the changelog so people know they should change their old entries when upgrading.

Reply:
We're still going to need parsing for other types of resources that have not been converted to the array format yet. Maybe for now we just add a comment like:

<pre>
// TODO: replace this method with a deprecation warning when all configs have been converted to arrays
</pre>

...and then we can phase it out gradually?

Reply:
Sounds fine for me